feat(ras-newsletter): Newsletters list progressive disclosure

Takshil-Kunadia opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Closes https://app.asana.com/0/1208970929679778/1208759817180346/f.

The Newsletters list displayed on the front-end to users to subscribe to during signup needed to be handled more gracefully. As per the discussion on ticket, this PR adds a "See all" button on the newsletter subscribe lists which when clicked displays the entire newsletter list. A setting is added in the Audience Configuration settings page in the "Newsletter Subscription Lists" section to set the number of initial newsletter to be displayed be default, while the rest of the newsletters hides behind the "See all" button.

Setting:-
<img width="482" alt="Screenshot 2025-04-05 at 7 38 21 PM" src="https://github.com/user-attachments/assets/4fb95636-e21a-442a-afc3-eac2ff5fa8c6" />

Front-end:-
<img width="483" alt="Screenshot 2025-04-05 at 7 36 04 PM" src="https://github.com/user-attachments/assets/e6af9a9d-481a-4d0e-b3c7-0313fb1d955d" />



### How to test the changes in this Pull Request:

1. Enable & Configure RAS and also enable the Present newsletter signup after checkout and registration in Audience settings.
2. Go to Newsletter > Settings and add a few Subscription list items with title & description.
3. Set the Initial list size to two or any number.
4. Log in as a guest window and click on Sign In, fill up the form.
5. Observe the newsletter list displaying the number of newsletter as set in the config and show an extra item partially.
6. Click on "See all" to display the entire list.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

Takshil-Kunadia committed:
feat(newsletters): add disclosure button for newsletter signup ui

Takshil-Kunadia committed:
feat(newsletters): add backend setting for initial list size

Takshil-Kunadia committed:
feat(newsletters): remove obvious comments

Takshil-Kunadia committed:
feat(newsletters): ui fixes and improve max height calculation for newsletter items and

thomasguillot committed:
fix: gap height

thomasguillot committed:
feat: add gradient to newsletter-list-container

thomasguillot committed:
fix: only display gradient when newsletter-list-container has a max-height

thomasguillot committed:
feat: update arrow-right icon to match wordpress/icons

thomasguillot commented:
@Takshil-Kunadia I pushed some very small changes:

* Reduced the gap to 12px
* Updated the arrow-right icon to match wordpress/icons
* Added a white to transparent gradient on the half-visible newsletter

Overall, this isn’t quite what I had in mind with the max-height. I was hoping to avoid a potentially super tall modal, especially after all the newsletters have loaded. That said, I really like what you’ve done, so let’s go with it 👍

✅ Design

leogermani commented:
Works as expected, but I have two comments.

First, the "See all" button is still visible after I expanded the list (see screenshot). (Maybe it wasn't at first, but became visible again after scrolling down and up a few times)

![image](https://github.com/user-attachments/assets/60b139ab-3a95-41f4-a353-bad012560199)

The help text under the new option could be more descriptive. Suggestion:

"Number of newsletters initially visible during signup. Additional newsletters will be hidden behind a 'See all' button."

cc @thomasguillot

Takshil-Kunadia committed:
fix(newsletters): update "See all" button logic

Takshil-Kunadia committed:
feat(newsletters): update setting help text

Takshil-Kunadia commented:
@leogermani Thanks for taking a look! 🙇 
- About the see all button issue, I wasn't able to replicate it on mac safari/chrome (I am yet to test the same on windows env). Regardless I have modified that code to remove the button entirely instead of just hiding it.
- Updated the help text ✅

leogermani commented:
Everything works fine!

But I still think the admin UI needs some loce. Pinging @thomasguillot  here:

![image](https://github.com/user-attachments/assets/b2b72991-2098-4cf1-bd0e-4a264726f066)

The "initial list size" is taking a lot of space. And you have that huge input to enter a tiny number. The list of Newsletters is moved down far from the section title... and it's not clear that they are part of the same section anymore...

Another idea: Limit the number of lists displayed to 2 minimum and only show that option if there are more than 2 lists available

@thomasguillot can you give us some directions here?

thomasguillot committed:
feat: use rangecontrol

thomasguillot committed:
feat: reorganise newsletter signup card

thomasguillot commented:
![update](https://github.com/user-attachments/assets/0e657cb1-2f9e-44f3-8f04-2e0c38fa2567)

@leogermani I've reorganised this whole section.

I've also replaced the input with a `RangeControl`. Right now it ranges between 1 and 10 which I think should be sufficent.

leogermani reviewed (approved):
Beautiful!
